pkg/migrations/mls/20240829001344_serial-ids.up.sql

```diff
@@ -0,0 +1,44 @@
+CREATE FUNCTION insert_group_message(group_id BYTEA, data BYTEA, group_id_data_hash BYTEA)
+	RETURNS SETOF group_messages
+	AS $$
+BEGIN
+	-- Ensures that the generated sequence ID matches the insertion order
+	-- Only released at the end of the enclosing transaction - beware if called within a long transaction
+	PERFORM
+		pg_advisory_xact_lock(hashtext('group_messages_sequence'));
```

Comment:
With the way this is queried, do we actually care that the `group_messages` table IDs are _globally_ monotonically increasing. What if we made the lock key the `group_id` instead?

Unlike with replication, no one is querying across multiple groups.

Same goes for welcome messages.

We could get away with doing the same for inbox logs too, even if there is a bit of a global namespace for those because of the address mapping. But that doesn't rely on the `sequence_id` being monotonically increasing, so I think it's fine to have that lock be scoped as well.